packages/application-shell/src/components/user-settings-menu/user-settings-menu.js

```diff
@@ -40,28 +40,40 @@ UserAvatar.propTypes = {
   gravatarHash: PropTypes.string.isRequired,
   firstName: PropTypes.string,
   lastName: PropTypes.string,
+  email: PropTypes.string,
   // Injected
   handleMouseOver: PropTypes.func.isRequired,
   handleMouseOut: PropTypes.func.isRequired,
   isMouseOver: PropTypes.bool.isRequired,
 };
 const UserAvatarWithHoverState = withMouseOverState(UserAvatar);
 
-export default class UserSettingsMenu extends React.PureComponent {
+export class UserSettingsMenu extends React.PureComponent {
   static displayName = 'UserSettingsMenu';
 
   static propTypes = {
     firstName: PropTypes.string,
     lastName: PropTypes.string,
+    email: PropTypes.string,
     gravatarHash: PropTypes.string.isRequired,
+
+    // withRouter
+    location: PropTypes.shape({
+      pathname: PropTypes.string.isRequired,
+    }).isRequired,
```

Comment:
You can also pass the location from the top not having to `withRouter` injecting a ton of other props?

Reply:
Yeah I can do that, I asked you guys what you think https://github.com/commercetools/merchant-center-application-kit/pull/151#discussion_r237447918

Reply:
Sorry didn't see that.